### PR TITLE
Reformat JSON output.

### PR DESCRIPTION
**Pull request by michaelgerakis**
This changes the JSON output of lighthouse to be more straightforward to parse and handle.

Here's a look at the overall structure:

#### overall + audits
image

#### aggregations
image

#### best practices, mix of audits + comingSoon placeholders
image

----------------------------

edit: paulirish put this description here. :)

### CODE REVIEW

**Comment by paulirish** [comment]
Looks great. The new JSON output feels on point.

Anyone else have comments on the format and structure of the output?

**Comment by brendankenny** [comment]
did you mean to commit outputs.json?

File: lighthouse-core/closure/typedefs/Aggregation.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ AggregationResultItem.prototype.name;
 /** @type {string} */
 AggregationResultItem.prototype.description;
 
-/** @type {!Array<!AuditResult>} */
+/** @type {!Array<!String>} */
```
Comment (brendankenny) [inline lighthouse-core/closure/typedefs/Aggregation.js]: lowercase `string`

**Comment by brendankenny** [comment]
@michaelgerakis can you give a motivation for the change? It looks fine so far, but it's not clear why you want to change it.

**Comment by michaelgerakis** [comment]
If someone is working with the results programmatically, they usually don't care about the aggregations. Currently, the only way to get to audits is to loop through all aggregations and then loop through its audits. If you already know you want one specific audit, then an object of audits is much more convenient.

**Comment by brendankenny** [comment]
ah, great, makes perfect sense

**Comment by brendankenny** [comment]
LGTM

**Comment by paulirish** [comment]
Here's a look at the overall structure:

#### overall + audits
image

#### aggregations
image

#### best practices, mix of audits + comingSoon placeholders
image

**Comment by paulirish** [comment]
Will keep open for a little longer in case anyone else has feedback on the output. 

We will likely not make major breaking changes to the JSON like this again.

**Comment by paullewis** [comment]
I'm good with this.

**Comment by michaelgerakis** [comment]
@yangchenyun and/or @zhaoz might be interested in this as well.

**Comment by hsablonniere** [comment]
I'm currently playing with lighthouse to test if we could use it to produce local custom reports and this is great for us :smile:

**Pull request merged**